src/testing/mocks.ts

```diff
@@ -21,71 +20,72 @@ import { noop } from '@utils';
 import { buildEvents } from '../compiler/events';
 import { createConfigFlags } from '../cli/config-flags';
 
-// TODO(STENCIL-486): Update `mockValidatedConfig` to accept any property found on `ValidatedConfig`
 /**
  * Creates a mock instance of an internal, validated Stencil configuration object
- * @param sys an optional compiler system to associate with the config. If one is not provided, one will be created for
  * the caller
+ * @param overrides a partial implementation of `ValidatedConfig`. Any provided fields will override the defaults
+ * provided by this function.
  * @returns the mock Stencil configuration
  */
-export function mockValidatedConfig(sys?: CompilerSystem): ValidatedConfig {
-  const baseConfig = mockConfig(sys);
+export function mockValidatedConfig(overrides: Partial<ValidatedConfig> = {}): ValidatedConfig {
```

Comment:
I think moving away from `sys` as a 'special' param here is a good example of making something more general and clear while at the same time preserving all of the functionality it had before (like if you want to supply a special `sys` param it's just a matter of `mockValidatedConfig({ sys })` 

it's a good pattern is what I'm saying 😄